spec/unit/module_spec.rb

```diff
@@ -347,6 +347,18 @@
       expect(mod.unmet_dependencies).to eq([])
     end
 
+    it "should throw an error if invalid dependencies are specified" do
+      expect {
+        PuppetSpec::Modules.create(
+          'foobar',
+          @modpath,
+          :metadata => {
+            :dependencies => ""
+          }
+        )
+      }.to raise_error(Puppet::Module::MissingMetadata)
```

Comment:
Since MissingMetadata is used elsewhere, could you check for the correct occurrence of the error with something like
```
      }.to raise_error(
        Puppet::Module::MissingMetadata, 
        /dependencies in the file metadata.json of the module foobar must be an array, not/)
```